Plugins/NativeCamera/Source/Apple/CameraDevice.mm

```diff
@@ -20,10 +20,24 @@
 
 @class CameraTextureDelegate;
 
-@interface CameraTextureDelegate : NSObject <AVCaptureVideoDataOutputSampleBufferDelegate>
+struct VideoRotationAngle
 {
-    @public AVCaptureVideoOrientation VideoOrientation;
+    enum Enum
+    {
+        LandscapeRight,
+        LandscapeLeft,
+        Portrait,
+        PortraitUpsideDown
+    };
+};
 
+@interface CameraTextureDelegate : NSObject <AVCaptureVideoDataOutputSampleBufferDelegate, NSObject>
+{
+#if (__MAC_OS_X_VERSION_MIN_REQUIRED >= 140000 || __IPHONE_OS_VERSION_MIN_REQUIRED >= 170000)
+    @public VideoRotationAngle::Enum VideoRotation;
+#else
+    @public AVCaptureVideoOrientation VideoOrientation;
```

Comment:
Can we just always use the new independent VideoRotationAngle enum, and for the older code path just convert to this type? Then it seems like we don't have to duplicate nearly as much code.

Reply:
Indeed, much better. Fixed.